include/exadg/postprocessor/time_control.cpp

```diff
@@ -0,0 +1,127 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+// C/C++
+#include <limits>
+
+// deal.II
+#include <deal.II/base/exceptions.h>
+
+// ExaDG
+#include <exadg/postprocessor/time_control.h>
+
+
+TimeControl::TimeControl()
+  : EPSILON(1.0e-10),
+    reset_counter(true),
+    is_active(false),
+    start_time(std::numeric_limits<double>::max()),
+    trigger_every_time_interval(-1.0),
+    trigger_every_time_steps(std::numeric_limits<unsigned int>::max()),
+    counter(0),
+    last_timestep_number_to_evaluate(dealii::numbers::invalid_unsigned_int)
+{
+}
+
+void
+TimeControl::reinit(bool const         is_active_,
+                    double const       start_time_,
+                    double const       trigger_every_time_interval_,
+                    unsigned int const trigger_every_time_steps_)
+{
+  is_active = is_active_;
+
+  if(is_active_)
+  {
+    if(trigger_every_time_steps_ == std::numeric_limits<unsigned int>::max() ||
+       trigger_every_time_steps_ == 0)
+    {
+      AssertThrow(trigger_every_time_interval_ > 0.0 &&
+                    trigger_every_time_interval_ < std::numeric_limits<double>::max(),
+                  dealii::ExcMessage("Input parameters are in conflict."));
+    }
+    else
+    {
+      AssertThrow(trigger_every_time_interval_ < 0.0 ||
+                    trigger_every_time_interval_ >= std::numeric_limits<double>::max(),
+                  dealii::ExcMessage("Input parameters are in conflict."));
+    }
+  }
+
+  start_time                  = start_time_;
+  trigger_every_time_interval = trigger_every_time_interval_;
+  trigger_every_time_steps    = trigger_every_time_steps_;
+}
+
+// If reinit() is not called needs_evaluation() will always evaluate to false
+bool
+TimeControl::needs_evaluation(double const time, unsigned int const time_step_number) const
+{
+  if(is_active == false)
+    return false;
+
+  if(time_step_number < 0) // steady problem (time_step_number = -1)
```

Comment:
I would change the comment here: "Steady problems are defined by a negative time step number"

Reply:
In my opinion we should be explicit about this anyway, see:
https://github.com/exadg/exadg/pull/263
If you agree, we should merge this PR before this one.